### PR TITLE
Adds all latest GEM datasets.

### LINKED ISSUE #80: Add GEM coal plan data
This is a nice exercise and can be similarly achieved with PR https://github.com/PyPSA/powerplantmatching/pull/79
Let me know if someone wants to tackle that. We have a data interface.

### LINKED ISSUE #81: Add GEM geothermal data
This is a nice exercise and can be similar achieved with PR https://github.com/PyPSA/powerplantmatching/pull/79

### PR DESCRIPTION
**Pull request by Tomkourou**
Closes #80 and closes #81. 

## Change proposed in this Pull Request

Adds all the latest datasets from GEM. Including Geothermal, Wind, Solar, Gas & Coal. 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other

## Checklist

- [x] I have added a note to release notes `doc/release_notes.rst`.
- [x] I have used `pre-commit run --all` to lint/format/check my contribution
- [x] I have documented the effects of my code changes in the documentation `doc/`.
- [x] I have adjusted the docstrings in the code appropriately.

### CODE REVIEW

**Comment by Tomkourou** [comment]
Whilst updating GCPT I noticed that the `contributing.rst` has wrong information. It suggests running the following function as the last step: 


```python 
pm.powerplants(update_all=True)
```
However this now does nothing as `update_all` has been deprecated. As seen below. 

`WARNING:powerplantmatching.collection:The following arguments were deprecated and are being ignored: {'update_all'}`

@FabianHofmann  what do you think about this? Would be good to update the `contributing.rst` doc.

**Comment by FabianHofmann** [comment]
Hey @Tomkourou, indeed we should fix this. It should be `update = True`. Would you be so kind and make a PR?

**Comment by FabianHofmann** [comment]
And thanks btw. Let me know when this is ready to review :)

**Comment by Tomkourou** [comment]
> And thanks btw. Let me know when this is ready to review :)

Ready to go :)

File: powerplantmatching/data.py [hunk 1]
```diff
@@ -1582,6 +1582,256 @@ def IRENASTAT(raw=False, update=False, config=None):
...
+            lon=df["lon"].apply(pd.to_numeric, errors="coerce"),
+        )
+        .pipe(lambda x: x[df.columns.intersection(config.get("target_columns"))])
+        .pipe(config_filter, config)
+    )
+
+    return df_final
+
+
+def GGtPT(raw=False, update=False, config=None):
```
Comment (pz-max) [inline powerplantmatching/data.py:1662]: I see why you named it "GGtPT", but what do you think about renaming it "GGTPT" instead? 
Would require changes across the PR wherever GGtPT was mentioned

File: powerplantmatching/package_data/config.yaml [hunk 2]
```diff
@@ -23,11 +23,13 @@ matching_sources:
...
   - GGPT
+  - GCPT
 
 fully_included_sources:
   # Make individual queries for the datasets
   - BEYONDCOAL
   - GGPT
+  - GCPT
   - OPSD: Country not in ['Switzerland', 'Italy', 'Spain', 'Norway', 'Austria']
   - JRC: Country not in ['Switzerland', 'Albania', 'United Kingdom', 'Norway']
```
Comment (pz-max) [inline powerplantmatching/package_data/config.yaml:34]: I think you need to add the other sources as well to be tested. This contributing section described tests:
https://powerplantmatching.readthedocs.io/en/latest/contributing.html

**Review by pz-max — COMMENTED** [review]
Really cool! :) Before delving into code optimization some comments in this review.

Additional to the comments, do you think it might make sense to report how the powerplant capacity per carrier changed? Like comparing pre-contribution capacity per carrier vs after contribution.

File: powerplantmatching/data.py (hunk 1 above)
Comment (Tomkourou) [reply powerplantmatching/data.py:1662]: Main reason I went with GGtPT was to stay consistent with how GEM call their datasets. See here https://globalenergymonitor.org/projects/global-geothermal-power-tracker/ . I would suggest we stay with their naming conventions.

**Comment by Tomkourou** [comment]
> Really cool! :) Before delving into code optimization some comments in this review.
> 
> Additional to the comments, do you think it might make sense to report how the powerplant capacity per carrier changed? Like comparing pre-contribution capacity per carrier vs after contribution.

Here is the analysis with the standard European target_country list:
new dataset: ENTSOE, GEO, GPD, JRC, OPSD, BEYONDCOAL, WIKIPEDIA, GGPT, GCPT, GCPT, GGtPT, GSPT, GWPT

original dataset: ENTSOE, GEO, GPD, JRC, OPSD, BEYONDCOAL, WIKIPEDIA, GGPT

Technology | new dataset | original dataset | Percentage difference |
| ------------- | ------------- | ------------- | ------------- |
Bioenergy | 6048.6834 | 6067.6834 | -0.3%
Geothermal | 806 | 766 | 5.2 %
Hard Coal | 183210.371796 | 116945.593026 | 56.7%
Hydro | 214002.872826 | 214054.872826 | 0.0%
Lignite | 77853.297197 | 56778.921 | 37.1%
Natural Gas | 219722.632261 | 230497.597932 | -4.7%
Nuclear | 137371.798212 | 138391.798212 | -0.7%
Oil | 18454.470346 | 20862.942634 | -11.5%
Other | 3728.809314 | 3695.380957 | 0.9%
Solar | 25663.803 | 3778.903 | 579.1%
Waste | 2950.606334 | 2950.606334 | 0.0%
Wind | 208877.81 | 24401.58 | 756.0%

image

**Comment by pz-max** [comment]
Hi @Tomkourou, 
The code looks good. I just added nuclear and bioenergy here: https://github.com/Tomkourou/powerplantmatching/pull/1/files. (Please merge the PR if you think it is ok, so it will appear here too). As minor detail, I changed `GGtPT` to `GGTPT` for naming consistency. Most of our data names are by default in capital letter. Any dev will think here its a bug even though its named like GGtPT officially by GEM.

@FabianHofmann, in case you start reviewing, one minor remaining question. Given a `config.yaml` entry like:
```
GSPT:
  net_capacity: false
  reliability_score: 4
  fn: Global-Solar-Power-Tracker-January-2023.csv
  url: https://raw.githubusercontent.com/pz-max/gem-powerplant-data/main/Global-Solar-Power-Tracker-January-2023.csv
```
Should we add something like `aggregated_units: true`? (some powerplants like nuclear have the same name for different powerplant units. Does that lead to conflicts? Will `aggregated_units: true` help?). Probably we could test that @Tomkourou :+1:

File: powerplantmatching/data.py [hunk 3]
```diff
@@ -1582,6 +1582,365 @@ def IRENASTAT(raw=False, update=False, config=None):
...
+        "Retired year": "DateOut",
+        "Technology Type": "Technology",
+        "GEM phase ID": "projectID",
+    }
+
+    technology_dict = {
+        "PV": "NaN",
+        "Solar Thermal": "Steam Turbine",
+        "Assumed PV": "NaN",
+    }
```
Comment (pz-max) [inline powerplantmatching/data.py:1921]: Maybe change "NaN" to "PV".
Easier to filter out later on.
"Solar Thermal" could be renamed to "CSP"

Comment (pz-max) [reply powerplantmatching/data.py:1921]: Changing to this code:
```
    technology_dict = {
        "PV": "PV",
        "Solar Thermal": "CSP",
        "Assumed PV": "PV",
    }
 ```
 Would create the output for `df.Technology.unique()` as ["PV", "CSP"]

File: powerplantmatching/data.py [hunk 4]
```diff
@@ -1631,25 +1990,89 @@ def GGPT(raw=False, update=False, config=None):
...
+
+    RENAME_COLUMNS = {
+        "Project Name": "Name",
+        "Capacity (MW)": "Capacity",
+        "Latitude": "lat",
+        "Longitude": "lon",
+        "Start Year": "DateIn",
+        "Retired Year": "DateOut",
+        "GEM unit ID": "projectID",
+        "Country 1": "Country",
```
Comment (FabianHofmann) [inline powerplantmatching/data.py:2047]:
```suggestion
        "Country 1": "Country",
        "Technology Type": "Technology",
```

**Review by FabianHofmann — CHANGES_REQUESTED** [review]

**Review by FabianHofmann — APPROVED** [review]

**Comment by FabianHofmann** [comment]
thanks @Tomkourou! looks all very nice!

**Comment by FabianHofmann** [comment]
@Tomkourou one question: You added that the GEM datasets are not reporting net capacity. Do you remember where you got this information from? I cannot find it on the GEM page. Since PPM should report net capacity, this might lead to some issues if we do not rescale the capacities. Thanks!

**Comment by Tomkourou** [comment]
@FabianHofmann it's a good question. My understanding was that since most GEM datasets include multiple units for powerplants then they needed to be summed up in the process and that's why I categorised them as not reporting net capacity. 

Perhaps I misunderstood the meaning of net capacity though and it needs to be recategorised?

**Comment by FabianHofmann** [comment]
@Tomkourou thanks for your answer! Actually, the correct tag for that would be `aggregated_units`. But this is no problem at all, since the data is effectively treated as net capacity in the code, which then correct and makes things easier. I will just remove the `net_capacity` tags from the config.

**Pull request merged**